src/Sparrow/Json/BlittableJsonReaderObject.cs

```diff
@@ -45,6 +45,11 @@ public void WriteJsonTo(Stream stream)
         public BlittableJsonReaderObject(byte* mem, int size, JsonOperationContext context,
             UnmanagedWriteBuffer buffer = default(UnmanagedWriteBuffer))
         {
+            if(size == 0)
+                //otherwise SetupPropertiesAccess will throw because of the memory garbage
```

Comment:
This is in the hot code path, and we need to avoid throwing exceptions there.
Even the _code_ for throwing exceptions is expensive, and we want to move that to a dedicated method so we'll not blow up the size of this method and thus avoid inlining it.